core/src/main/java/org/apache/iceberg/hadoop/JdbcCatalog.java

```diff
@@ -0,0 +1,314 @@
+    dataSource.setUsername(properties.get(JDBC_CATALOG_USER));
+    dataSource.setPassword(properties.get(JDBC_CATALOG_PASSWORD));
+    this.queryRunner = new QueryRunner(dataSource);
+
+    try {
+      initializeCatalog();
+    } catch (SQLException throwables) {
+      throw new RuntimeIOException("Failed to initialize Jdbc Catalog!\n %s %s", throwables.getErrorCode(),
+              throwables.getMessage());
+    }
+
+  }
+
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    return new JdbcTableOperations(queryRunner, fileIO, catalogName, tableIdentifier);
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+    String tableName = tableIdentifier.name();
+    StringBuilder sb = new StringBuilder();
+
+    sb.append(warehouseLocation).append('/');
+    for (String level : tableIdentifier.namespace().levels()) {
+      sb.append(level).append('/');
+    }
+    sb.append(tableName);
+
+    return sb.toString();
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    TableOperations ops = newTableOps(identifier);
+    TableMetadata lastMetadata = ops.current();
+    try {
+      new JdbcTableDao(queryRunner, catalogName).delete(identifier);
+      if (purge && lastMetadata != null) {
+        CatalogUtil.dropTableData(ops.io(), lastMetadata);
+        FileSystem fs = Util.getFs(new Path(warehouseLocation), hadoopConf);
+        fs.delete(new Path(lastMetadata.location()), true /* recursive */);
+        LOG.info("Table {} data purged!", identifier);
+      }
+      return true;
+    } catch (SQLException | IOException e) {
+      LOG.error("Cannot complete drop table operation for {} due to unexpected exception {}!", identifier,
+              e.getMessage(), e);
+      throw new RuntimeIOException("Failed to drop table %s", identifier.toString());
+    }
+  }
+
+  public void logTables() {
+    try {
+      List<JdbcTable> tables = queryRunner.query("SELECT * FROM " + JdbcTableDao.TABLES_TABLE_NAME,
+              new BeanListHandler<>(JdbcTable.class));
+      if (tables.isEmpty()) {
+        LOG.info("No Table found!");
+      }
+      for (JdbcTable table : tables) {
+        LOG.info("Table:{} ,NS: {}", table.toTableIdentifier().toString(), table.getTableNamespace().toString());
+        LOG.warn("IS NULL:{} ", table.getTableNamespace() == null);
+      }
+    } catch (SQLException throwables) {
+      LOG.error("Failed to list tables!", throwables);
+    }
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+
+    JdbcNamespaceDao nsDao = new JdbcNamespaceDao(queryRunner, catalogName);
+    if (nsDao.isExists(namespace)) {
+      throw new NoSuchNamespaceException("Namespace %s does not exist!", namespace.toString());
+    }
+    JdbcTableDao tableDao = new JdbcTableDao(queryRunner, catalogName);
+    List<TableIdentifier> results = Lists.newArrayList();
```

Comment:
Is this List allocation necessary? It seems like it's discarded when results is reassigned from `tableDao.getAll(namespace)` below.